src/toolbar/toolbar-title.jsx

```diff
@@ -14,6 +14,10 @@ function getStyles(props, state) {
       fontSize: toolbar.titleFontSize,
       display: 'inline-block',
       position: 'relative',
+      textOverflow: 'ellipsis',
+      whiteSpace: 'nowrap',
+      overflow: 'hidden',
+      width: (props.text.length > 40) ? 150 : 'inherit',
```

Comment:
Yeah. I agree 40 and 50 are indeed magic numbers. I tried using flex it didn't help.

Thats because the `ToolbarTitle` becomes a part of the `ToolbarGroup` and depends on the number of components being included in the `ToolBar`. 'AppBar' has title as a property where as ToolbarTitle is a component in itself and its width will depend on the width of its parent element.

Correct me if I am wrong here.